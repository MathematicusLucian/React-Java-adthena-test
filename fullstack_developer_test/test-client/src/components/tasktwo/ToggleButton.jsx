import React from 'react';
import PropTypes from 'prop-types';

const ToggleButton = ({
  theme,
  onClick,
}) => {

  return (
    <button
      type="button"
      onClick={onClick}
    >
      Switch theme to <span>{theme}</span>
    </button>
  );
};

ToggleButton.propTypes = {
  theme: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

ToggleButton.defaultProps = {
  onClick: () => {},
};

export default ToggleButton;