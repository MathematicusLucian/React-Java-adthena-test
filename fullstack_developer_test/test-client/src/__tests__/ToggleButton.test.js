import React from 'react';
import { shallow } from "enzyme";
import ToggleButton from './../components/tasktwo/ToggleButton';

describe('ToggleButton', () => {

  let wrapper;
  const theme = "Dark";

  beforeEach(() => {
    const props = {};
    wrapper = shallow(<ToggleButton {...props} theme={theme}></ ToggleButton>);
  });

  it('should render correctly', () => {
    shallow(<ToggleButton theme={theme} />);
  });

  it('should render - shallow', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ToggleButton debug theme={theme} />)
    expect(component).toMatchSnapshot();
  });

  it('should render a button tag', () => {
    expect(wrapper.find('button').exists()).toBe(true);
  });

  it('should render a button tag - 2', () => {
    expect(wrapper.find("button").length).toEqual(1);
  });

  it('button text should read "Switch theme to Dark"', () => {
    expect(wrapper.find("span").first().text()).toBe("Dark");
    expect(wrapper.find("button").first().text()).toBe("Switch theme to Dark");
  });

});