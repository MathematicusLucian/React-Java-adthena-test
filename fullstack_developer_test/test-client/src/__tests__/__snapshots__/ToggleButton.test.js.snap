// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton should render correctly in "debug" mode 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Switch theme to 
  <span>
    Dark
  </span>
</button>
`;
